TombLib/TombLib/LevelData/IO/Prj2Loader.cs

```diff
@@ -405,8 +405,8 @@ private static bool LoadLevelSettings(ChunkReader chunkIO, ChunkId idOuter, stri
                                     for (int x = 0; x < levelTexture.FootStepSoundWidth; ++x)
                                     {
                                         byte textureSoundByte = chunkIO.Raw.ReadByte();
-                                        if (textureSoundByte > 15)
-                                            textureSoundByte = 15;
+                                        if (textureSoundByte > (byte)Enum.GetNames(typeof(TextureFootStep.Type)).Length)
+                                            textureSoundByte = (byte)Enum.GetNames(typeof(TextureFootStep.Type)).Length;
```

Comment:
This will fail, as you're assigning a value that's larger than the enum size.
Additionally, I wouldn't use the `Enum.GetNames()` method as I think it uses Reflection underneath.
Please add a value at the end of the enum called `Count` and use that one instead:
![image](https://github.com/MontyTRC89/Tomb-Editor/assets/20436882/e667985d-9d56-4ec3-a211-442b64dd31c0)

Then replace those 2 lines you edited with this:
```cs
byte maxSoundByte = (byte)(TextureFootStep.Type.Count - 1);

if (textureSoundByte > maxSoundByte)
	textureSoundByte = maxSoundByte;
```

Reply:
Thanks, indeed I forgot the - 1.
Just did what you asked, and edited `TextureFootStep.GetNames()` to avoid having `Count` appear in the dropdown.